File: back_end/java_version/anti_patterns/src/main/java/com/shipay/challenge/endpoints/registration/RegistrationService.java
package com.shipay.challenge.endpoints.registration;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class RegistrationService {

    @Autowired
    private CustomerRepository repository;

    public void save(Customer customer) {
        repository.save(customer);
    }
    
    public Customer getCustomerById(Long id) {
        return repository.findById(id).orElse(null);
    }
}

File: back_end/teahupoo/anti_patterns/Endpoints/Registration/RegistrationService.cs
namespace Shipay.Challenge.Endpoints.Registration
{
    public class RegistrationService
    {
        private readonly MyDbContext _context;

        public RegistrationService(MyDbContext context)
        {
            _context = context;
        }

        public void SaveCustomer(Customer customer)
        {
            _context.Customers.Add(customer);
            _context.SaveChanges();
        }

        public Customer GetById(int id)
        {
            return _context.Customers.Find(id);
        }
    }
}
